src/perl/bin/bulk_publish_wh_messages

```diff
@@ -0,0 +1,154 @@
+#! /usr/bin/env perl
+
+package bulk_publish_wh_messages;
+
+use Moose;
+use FindBin qw($Bin);
+use lib ( -d "$Bin/../lib/perl5" ? "$Bin/../lib/perl5" : "$Bin/../lib" );
+use Carp;
+
+use wtsi_clarity::util::bulk_publish_wh_messages;
+
+with 'MooseX::Getopt';
+
+has 'model'  => (
+  isa             => 'Str',
+  is              => 'ro',
+  required        => 1,
+);
+
+has 'process_url' => (
+  isa             => 'Str',
+  is              => 'ro',
+  required        => 0,
+);
+
+has 'project_name'  => (
+  isa               => 'Str',
+  is                => 'ro',
+  required          => 0,
+);
+
+my $self = bulk_publish_wh_messages->new_with_options();
+my $parameters = { model => $self->model };
+
+if ((defined $self->process_url and length $self->process_url) and (defined $self->project_name and length $self->project_name)) {
```

Comment:
Nothing wrong with doing it this way but for convenience you can set predicate methods on Moose attributes that return true if the attribute has been set:

```perl
package bulk_publish_wh_messages;

has 'process_url' => ( predicate => 'has_process_url' );
my $self = bulk_publish_wh_messages->new_with_options();

if ($self->has_process_url) { ... }
```